ketcher-autotests/tests/Templates/Template-Manipulations/Template-manipulations.spec.ts

```diff
@@ -0,0 +1,324 @@
+    /*
+    Test case: 1745
+    Description: Choose the 'Rotate Tool', select the structure and rotate it.
+    Select the structure and flip it horizontally with the 'Horizontal Flip' tool.
+    Select the structure and flip it vertically with the 'Vertical Flip' tool.
+    */
+    const anyAtom = 0;
+    await drawBenzeneRing(page);
+    await selectAtomInToolbar(AtomButton.Fluorine, page);
+    await clickOnAtom(page, 'C', anyAtom);
+    await page.keyboard.press('Control+a');
+    await pressButton(page, 'Vertical Flip (Alt+V)');
+    await takeEditorScreenshot(page);
+    await pressButton(page, 'Horizontal Flip (Alt+H)');
+  });
+
+  test('Save as *.mol file', async ({ page }) => {
+    /*
+    Test case: 1747
+    Description: Click the 'Save As' button, and click the 'Save' button.
+    Open the saved *.mol file and edit it in any way.
+    */
+    await openFileAndAddToCanvas('Molfiles-V2000/three-templates.mol', page);
+    const expectedFile = await getMolfile(page, 'v2000');
+    await saveToFile(
+      'Molfiles-V2000/three-templates-expected.mol',
+      expectedFile,
+    );
+    const METADATA_STRING_INDEX = [1];
+    const { fileExpected: molFileExpected, file: molFile } =
+      await receiveFileComparisonData({
+        page,
+        expectedFileName:
+          'tests/test-data/Molfiles-V2000/three-templates-expected.mol',
+        fileFormat: 'v2000',
+        metaDataIndexes: METADATA_STRING_INDEX,
+      });
+    expect(molFile).toEqual(molFileExpected);
+  });
+
+  test('Save as *.rxn file', async ({ page }) => {
+    /*
+    Test case: 1748
+    Description: Click the 'Save As' button and click the 'Save' button.
+    Open the saved *.rxn file and edit it in any way.
+    */
+    await openFileAndAddToCanvas('Rxn-V2000/templates-reaction.rxn', page);
+    const expectedFile = await getRxn(page, 'v2000');
+    await saveToFile('Rxn-V2000/templates-reaction-expected.rxn', expectedFile);
+    // eslint-disable-next-line no-magic-numbers
+    const METADATA_STRINGS_INDEXES = [2, 7, 43, 63];
+    const { file: RxnFile, fileExpected: RxnFileExpected } =
+      await receiveFileComparisonData({
+        page,
+        expectedFileName:
+          'tests/test-data/Rxn-V2000/templates-reaction-expected.rxn',
+        metaDataIndexes: METADATA_STRINGS_INDEXES,
+      });
+    expect(RxnFile).toEqual(RxnFileExpected);
+  });
+
+  test('If connect to a single bond with two atoms then replace double bond by single bond for the fusion', async ({
+    page,
+  }) => {
+    /*
+    Test case: EPMLSOPKET-15504
+    Add cyclopentadiene ring on canvas
+    Add another cyclopentadiene ring to a single bond with two atoms, where each atom is connected to any atom with a double bond
+    */
+    await drawCyclopentadieneRing(page);
+    await selectAtomInToolbar(AtomButton.Nitrogen, page);
+    await clickOnAtom(page, 'C', 0);
+    await clickOnAtom(page, 'C', 3);
+    await selectRing(RingButton.Cyclopentadiene, page);
```

Comment:
those lines are repeated in the next test, it may be better to wrap them in a function
    await selectAtomInToolbar(AtomButton.Nitrogen, page);
    await clickOnAtom(page, 'C', 0);
    await clickOnAtom(page, 'C', 3);
    await selectRing(RingButton.Cyclopentadiene, page);'

Reply:
done

Reply:
Done.